crates/sui-source-validation-service/src/lib.rs

```diff
@@ -62,16 +63,24 @@ pub enum PackageSources {
 pub struct RepositorySource {
     pub repository: String,
     pub branch: String,
-    pub paths: Vec<String>,
+    pub packages: Vec<Package>,
     pub network: Option<Network>,
 }
 
 #[derive(Clone, Deserialize, Debug)]
 pub struct DirectorySource {
-    pub paths: Vec<String>,
+    pub packages: Vec<Package>,
     pub network: Option<Network>,
 }
 
+#[derive(Clone, Deserialize, Debug)]
+pub struct Package {
+    pub path: String,
+    /// Optional object ID to watch for upgrades. For framework packages, this is an address like 0x2.
+    /// For non-framework packages this is an upgrade cap (possibly wrapped).
```

Comment:
These will both look the same in the TOML, how do you distinguish between the two?

Reply:
(I see in the follow-up that it doesn't really matter what it is because all you're looking for is an object that will change with each upgrade).